### PR TITLE
Use Agent UUID from machine id

### PR DESCRIPTION
**Pull request by nelsonkopliku**
With this PR we create anAgentID which is a UUID ceated from the content in `/etc/machine-id`.

The created UUID will be the same for the same machineID content given.

We have uuid and uniqueness of agent identifier.

Closes https://github.com/trento-project/trento/issues/425

### CODE REVIEW

**Review by dottorblaster — APPROVED** [review]

File: agent/discovery/collector/client.go [hunk 1]
```diff
@@ -64,10 +65,15 @@ func NewCollectorClient(config *Config) (*client, error) {
 
 	machineID := strings.TrimSpace(string(machineIDBytes))
 
+	agentID, err := uuid.NewRandomFromReader(strings.NewReader(machineID))
+	if err != nil {
```
Comment (fabriziosestito) [inline agent/discovery/collector/client.go:69]: Is the UUID Random?
In this case what happens if the agent or the machine is restarted?

We need to find a way to have deterministic IDs, e.g. namespacing could help

**Review by fabriziosestito — CHANGES_REQUESTED** [review]

File: agent/discovery/collector/client.go (hunk 1 above)
Comment (nelsonkopliku) [reply agent/discovery/collector/client.go:69]: From what I could see, this line returns always the same UUID given the same machineID.

I can double check.

Comment (rtorrero) [reply agent/discovery/collector/client.go:69]: I think this is a seed for a random number so yes, given the same seed the random number should be the same.

Comment (nelsonkopliku) [reply agent/discovery/collector/client.go:69]: I'll add a test for this.

```
Pseudocode

loop n times and call
uuid.NewRandomFromReader(strings.NewReader(machineID))

check all the uuids are the same
```

Would this help?

Comment (fabriziosestito) [reply agent/discovery/collector/client.go:69]: I would still use a namespace for this one rather than the random  uuid

Comment (nelsonkopliku) [reply agent/discovery/collector/client.go:69]: I;ll give it a try :wink:

File: agent/discovery/collector/client.go [hunk 2]
```diff
@@ -35,6 +36,8 @@ type Config struct {
 	CA            string
 }
 
+var UuidNamespace = uuid.Must(uuid.Parse("fb92284e-aa5e-47f6-a883-bf9469e7a0dc"))
```
Comment (nelsonkopliku) [inline agent/discovery/collector/client.go:39]: I do not really like it here, I see this at a higher level in our modules, but I am getting bitten by cyclic imports.
Gonna keep it here for now if that's okay. :innocent:

File: agent/discovery/collector/client.go (hunk 1 above)
Comment (nelsonkopliku) [reply agent/discovery/collector/client.go]: @fabriziosestito changed, please take a look.

File: agent/discovery/collector/client.go (hunk 2 above)
Comment (fabriziosestito) [reply agent/discovery/collector/client.go:39]: does moving this to `internal/uuid.go` helps? 
I would call it `TrentoNamespace` or something similar to  make it clear that is our defined namespace

**Review by fabriziosestito — APPROVED** [review]
Little nitpick on naming, but feel free to merge and address it later

File: agent/discovery/collector/client.go (hunk 2 above)
Comment (nelsonkopliku) [reply agent/discovery/collector/client.go:39]: Deal.

**Pull request merged**